code/modules/mob/living/carbon/carbon_update_icons.dm

```diff
@@ -335,3 +397,58 @@
 		. += "-HAIR_HIDDEN"
 
 	return .
+
+/obj/item/bodypart/r_leg/generate_icon_key()
+	RETURN_TYPE(/list)
+	. = ..()
+	if(left_leg_mask_key) // We do this so we can cache the versions with and without a mask, for when there's no left leg.
+		. += "-[left_leg_mask_key]"
+
+	return .
+
+/**
+ * This proc serves as a way to ensure that right legs don't overlap above left legs when their dir is WEST on a mob.
+ *
+ * It's using the `left_leg_mask_cache` to avoid generating a new mask when unnecessary, which means that there needs to be one
+ * for the proc to return anything.
+ *
+ * Arguments:
+ * * right_leg_icon_file - The icon file of the right leg overlay we're trying to apply a mask to.
+ * * right_leg_icon_state - The icon_state of the right leg overlay we're trying to apply a mask to.
+ * * image_dir - The direction applied to the icon, only meant for when the leg is dropped, so it remains
+ * facing SOUTH all the time.
+ *
+ * Returns the `/image` of the right leg that was masked, or `null` if the mask didn't exist.
+ */
+/obj/item/bodypart/r_leg/proc/generate_masked_right_leg(right_leg_icon_file, right_leg_icon_state, image_dir)
+	if(!left_leg_mask_key || !left_leg_mask_cache[left_leg_mask_key] || !right_leg_icon_file || !right_leg_icon_state)
```

Comment:
I highly considered it, until remembering it can be null too. And since I wasn't confident with how it works, I just didn't do it just to not make it so people expect this to be something else.

Reply:
RETURN_TYPE() allows nulls too, its common behavior for a proc to either return 1 type or null.

Reply:
Makes sense, I'm just used to having to specify returning undefined in typescript so it makes me a little wary, and I somewhat wish I could specify it can return null, if possible.

Reply:
It's only purpose is to allow things like 
thing.return_list().Join()

I wish we had ACTUAL return types 😭